src/plugins/management/public/components/management_app/management_app.tsx

```diff
@@ -79,8 +82,19 @@ export const ManagementApp = ({ dependencies, history }: ManagementAppProps) =>
   );
 
   useEffect(() => {
+    // If workspace is enabled and user has entered workspace, hide advance settings and dataSource menu
+    if (capabilities.workspaces.enabled && workspaces.currentWorkspace$.getValue()) {
+      const opensearchDashboardsSection = dependencies.sections
+        .getSectionsEnabled()
+        .find((section) => section.id === ManagementSectionId.OpenSearchDashboards);
+      const disabledApps = opensearchDashboardsSection?.apps.filter(
+        (app) => app.id === 'settings' || app.id === 'dataSources'
+      );
+      disabledApps?.forEach((app) => app.disable());
+    }
+
     setSections(dependencies.sections.getSectionsEnabled());
-  }, [dependencies.sections]);
+  }, [dependencies.sections, capabilities, workspaces]);
```

Comment:
```suggestion
  }, [dependencies.sections.getSectionsEnabled(), capabilities.workspaces.enabled, workspaces.currentWorkspace$.getValue()]);
```
The pointer will not change I suppose, use the honored values as dependency should be better.